controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -187,7 +188,16 @@ public OperationContext createStreamContext(String scopeName, String streamName,
                         Futures.completeOn(checkScopeExists(scope, context, executor)
                                 .thenCompose(exists -> {
                                     if (exists) {
-                                        // Create stream may fail if scope is deleted as we attempt to create the stream under scope. 
+                                        checkScopeInDeletingTable(scope, context, executor)
+                                                .thenCompose(exist -> {
+                                                    if (exist) {
+                                                        throw new IllegalArgumentException("Scope already in deleting state: " + scope);
```

Comment:
Throwing an IllegalArgumentException does not seem to be the right thing to do. How/where is this being handled?

Reply:
the exception thrown from here is now changed.